frontend/src/api/network/notebooks.ts

```diff
@@ -256,8 +258,14 @@ export const updateNotebook = (
   data.notebookId = existingNotebook.metadata.name;
   const notebook = assembleNotebook(data, username);
 
+  const container = existingNotebook.spec.template.spec.containers[0];
+
   // clean the envFrom array in case of merging the old value again
-  existingNotebook.spec.template.spec.containers[0].envFrom = [];
+  container.envFrom = [];
+  // clean the resources, affinity and tolerations for GPU
+  existingNotebook.spec.template.spec.tolerations = [];
+  existingNotebook.spec.template.spec.affinity = {};
```

Comment:
We need to clone this. I missed it during the rapid development of DSG. You're mutating likely state from the context here -- we can't trust variable sources of what is passed to us, let us avoid mutating those.